source/Game.cpp

```diff
@@ -1,199 +1,104 @@
 #include"Game.hpp"
 
-#include<sstream>
-#include<iostream>
-#include<cctype>
-#include<string>
-#include"Error.hpp"
+#include"ActionOption.hpp"
+#include"EnhanceOption.hpp"
 
-namespace dungeon {
-
-Game::Game(): isEnd(false) {
-  initialize();
-}
+#include<chrono>
+#include<utility>
 
-void Game::welcome() {
-  ios.clear();
-  ios.print("hello, world");
-  ios.pause();
-}
+namespace dungeon {
 
-void Game::selectClass() {
-  ios.clear();
-  ios.print("Select your Class");
-  ios.print("A. Saber");
-  ios.print("B. Archer");
-  ios.print("C. Lancer");
-  // ios.print("D. Rider");
-  // ios.print("E. Caster");
-  // ios.print("F. Assassin");
-  ios.print("G. Berserker");
-  switch (ios.inputChar()) {
-  case 'A':
-    mainChar.type = Class::saber; break;
-  case 'B':
-    mainChar.type = Class::archer; break;
-  case 'C':
-    mainChar.type = Class::lancer; break;
-  // case 'D':
-  //   mainChar.type = Class::rider; break;
-  // case 'E':
-  //   mainChar.type = Class::caster; break;
-  // case 'F':
-  //   mainChar.type = Class::assassin; break;
-  case 'G':
-    mainChar.type = Class::berserker; break;
+Game::Game(): rndGen(std::chrono::steady_clock::now().time_since_epoch().count()), map(rndGen) {
+  ios.clear(), ios.welcome();
+  auto name = ios.askMainCharName();
+  switch (ios.askMainCharClass()) {
+  case ClassType::SABER:
+    mainChar = std::make_unique<CharacterClass<ClassType::SABER>>(name); break;
+  case ClassType::ARCHER:
+    mainChar = std::make_unique<CharacterClass<ClassType::ARCHER>>(name); break;
+  case ClassType::LANCER:
+    mainChar = std::make_unique<CharacterClass<ClassType::LANCER>>(name); break;
+  case ClassType::BERSERKER:
+    mainChar = std::make_unique<CharacterClass<ClassType::BERSERKER>>(name); break;
   default:
-    throw TypeError();
+    std::unreachable();
```

Comment:
This should be an recoverable Invalid Input error instead of unreachable(?) though IDK what you will implement and return in askMainCharName. But don't let the program crash just because a user typo causing wrong input :hankey:

Reply:
I thought typo handling should be implement in IO system so I put unreachable here.
IO system should re-ask until a valid input.